attest/verifier/types/src/verification.rs

```diff
@@ -27,16 +27,108 @@ pub struct DcapEvidence {
     pub report_data: EnclaveReportDataContents,
 }
 
-#[derive(Clone, Oneof)]
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub enum EvidenceKind {
-    #[prost(message, tag = "4")]
+    Epid(VerificationReport),
     Dcap(prost::DcapEvidence),
 }
 
-#[derive(Clone, Message)]
-pub struct EvidenceMessage {
-    #[prost(oneof = "EvidenceKind", tags = "4")]
-    pub evidence: Option<EvidenceKind>,
+// We default to `EvidenceKind::Epid` as that was the original kind of
+// `VerificationReport`.
+impl Default for EvidenceKind {
+    fn default() -> Self {
+        EvidenceKind::Epid(Default::default())
+    }
+}
+
+impl From<VerificationReport> for EvidenceKind {
+    fn from(report: VerificationReport) -> Self {
+        EvidenceKind::Epid(report)
+    }
+}
+
+impl From<prost::DcapEvidence> for EvidenceKind {
+    fn from(evidence: prost::DcapEvidence) -> Self {
+        EvidenceKind::Dcap(evidence)
+    }
+}
+
+// The first tag for a `VerificationReport`, this needs to match that defined
+// in `VerificationReport`.
+const TAG_VERIFICATION_REPORT_FIRST: u32 = 1;
+
+// The last tag for a `VerificationReport`, this needs to match that defined
+// in `VerificationReport`.
+const TAG_VERIFICATION_REPORT_LAST: u32 = 3;
+
+// The tag to indicate a `DcapEvidence` variant. For backwards compatibility
+// this must be outside the range of tags used by `VerificationReport`.
+const TAG_DCAP_EVIDENCE: u32 = 4;
+
+/// In order to make `EvidenceKind` backwards compatible with the previous
+/// logic which would send `VerificationReport`'s, the protobuf tags that
+/// correspond to tags defined in `VerificationReport` will decode to the
+/// `EvidenceKind::Epid()` variant. Tags outside of a `VerificationReport`'s
+/// will be treated as other variants.
+impl Message for EvidenceKind {
```

Comment:
Curious on thoughts to this approach. I think the previous plan of:
1. Try decoding as `EvidenceMessage`
2. If it's `None`, then try decoding as `VerificationReport`
This allows consumers of `EvidenceKind` and the protobufs that may come along, to always try to decode as `EvidenceKind` and not worry as much about the nitty gritty.

Reply:
changed implementation, to have dedicated methods on the `EvidenceKind` so that it's less likely someone tries to use this serialization method in other places where it's not intended